content/news/2018-07-30-navpi-memory-management-v1-0-9-release-notes.md

```diff
@@ -0,0 +1,83 @@
+---
+layout: news
+generated_by_cms: true
+title: NavPi Memory Management & v1.0.9 Release Notes
+author: zanuka
+date: '2018-07-30T10:39:01+12:00'
+feature_image: /images/uploads/navpi3.jpg
+categories:
+  - Nav News
+  - NavPi
+---
+## NavPi Memory Management
+
+As most veteran NavPi users know, there has been a significant issue with the memory resources being completely maxed out while running the NavCoin daemon. This has resulted in the NavPi’s default Chromium WebUI being nearly unusable and has also caused the daemon to crash on new installations and upgrades.  Advanced users of the NavPi have tweaked their systems for optimised performance, but for new users the initial setup and configuration mods are most certainly daunting tasks. It’s been a frustrating issue for everyone and the core team has been focused on solving the problem by engineering both short and long term solutions.
+
+### The problem; memory is maxed
+
+When the NavCoin daemon starts, it loads the entire chainstate database in memory. This causes a large spike in memory usage until that operation is complete. Existing users have reported that after updating to the latest version of NavCoin, the memory spike can cause the Chromium WebUI to be nearly unusable in addition to the navcoind daemon simply crashing. 
+
+### The short term solution; add swap space
+
+Creating dedicated swap space was the key to resolving the memory issue, and three approaches; zram swap, usb swap, and sd swap.
+
+### What is the function of swap space?
+
+The swap space is an area on the Raspberry Pi’s SD card (or additional USB drive) intended as a place the operating system can store data temporarily.  
+
+### How does NavPi use swap space?
+
+NavPi uses the swap space to increase the amount of information it can process “in memory” while syncing the blockchain and performing other essential operations.
+
+### ZRAM swap
+
+The initial approach to the problem was to enable ZRAM on the NavPi as a temporary fix. For many users it seemed to help, but during the core team’s testing cycles there were still resource issues on the initial load of the chain state in memory. 
+
+* [How to enable ZRAM swap on NavPi](https://info.navcoin.org/knowledge-base/enable-zram-navpi/) (NavCoin Knowledge Base)
+
+### SD swap
+
+As another temporary solution, we created an article on the Knowledge Base demonstrating how to enable the swap space on the SD card of the NavPi. The steps in the article were based on an article written by Justin Ellingwood of Digital Ocean and tested successfully on NavPi v1.0.8. While this approach did result in added stability for both the NavCoin daemon and the Chromium WebUI, it is not our recommended approach long-term.
```

Comment:
Adding a reason as to why it isn't recommended wouldn't go amiss here I think
"...it is not our recommended approach long-term as this will cause the NavPi's SD Card to wear out faster."

Reply:
fixed